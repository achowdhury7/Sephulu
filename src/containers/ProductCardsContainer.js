import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { filterActions, productActions } from '../actions'
import { ProductCard } from '../components'


class ProductCardsContainer extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    const listItems = this.props.productIds.map(id => (
      <li className="product-list-item" key={id}><ProductCard productId={id} /></li>
    ))
    return (
      <ul className="product-list">{listItems}</ul>
    )
  }
}

const mapDispatchToProps = dispatch => 
  ({
    actions: bindActionCreators(Object.assign(
      {},
      filterActions,
      productActions),
      dispatch
    )
  })

const mapStateToProps = state => ({
  productIds: Object.keys(state.app.products.byId)
})


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProductCardsContainer)

