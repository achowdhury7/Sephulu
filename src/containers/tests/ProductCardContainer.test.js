import React from 'react'
import { shallow, mount, render } from 'enzyme'

import { ProductCardsContainer } from '../ProductCardsContainer'

test('render product cards container', () => {
  const mockProductIds = ['1', '2', '3']
  const wrapper = shallow(
    <ProductCardsContainer productIds={mockProductIds} isFetching={false} />
  )
  expect(wrapper).toMatchSnapshot()
})