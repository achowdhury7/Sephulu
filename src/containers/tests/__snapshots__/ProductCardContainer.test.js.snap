// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product cards container 1`] = `
<ul
  className="product-list"
>
  <li
    className="product-list-item"
    key="1"
  >
    <Connect(ProductCard)
      productId="1"
    />
  </li>
  <li
    className="product-list-item"
    key="2"
  >
    <Connect(ProductCard)
      productId="2"
    />
  </li>
  <li
    className="product-list-item"
    key="3"
  >
    <Connect(ProductCard)
      productId="3"
    />
  </li>
</ul>
`;
